scripts/pylib/twister/twisterlib/reports.py

```diff
@@ -349,7 +349,9 @@ def json_report(self, filename, version="NA", platform=None):
             suite['testcases'] = testcases
 
             if instance.recording is not None:
-                suite['recording'] = instance.recording
+                r = {}
+                r[instance.testsuite.id] = instance.recording
```

Comment:
this is  an extra object level introduced which is not needed inside of the testsuite object, but might be imprtant for whatever storage will be used to upload into.
imo it is better to deal with these conflicts at the upload point where constraints are known.
for example, just renaming `recording` to `recording__foo_test_suite_name` there on upload.

Reply:
I'd rather have this happen at the source, so there is no ambiguity and minimal need to post-process the data in custom ways...

Reply:
at the source level of the json there is already no ambiguity as the objects and names in this document are hierarchical.
the ambiguity might happen when these documents are transformed and stored with some arbitrary mapping schema applied, it might be split to different indexes or not, flattened, nested, etc. - it is up to the user how to deal with it better.
even `instance.testsuite.id` might be not unique, isn't it ?

Reply:
> at the source level of the json there is already no ambiguity as the objects and names in this document are hierarchical.

I mean, if the usage is going to require manipulating the data, let's just do it at the source and be done with it andguarantee  it is consistent where ever you use it.

Reply:
I don't understand the need for this change. I think for uploading you already have to do some processing? E.g. the entries of test suites are not identified by the "names". "Name" is just one entry within "testsuite", and status is at the same level. Following the logic, we would have to create a dictionary out of every field and give the test name as a key? This seems to be counter productive: less clear output and more redundant wording. 
Matrics are also just strings now. If one would like to track a particular value, e.g. # cycles they would anyhow need to parse the report to get the key - value pairs.
BTW please also note that "name" != "identifier" i.e.
```
"name":"tests/benchmarks/latency_measure/benchmark.kernel.latency.timeslicing",
...
recording":{
                "benchmark.kernel.latency.timeslicing":[
                    {
``` not sure which one is used for tracking